app-ios/tutanota/Sources/Contacts/ContactsSynchronization.swift

```diff
@@ -22,281 +29,259 @@ private let ALL_SUPPORTED_CONTACT_KEYS: [CNKeyDescriptor] = [
 
 /// Handles synchronization between contacts in Tuta and contacts on the device.
 class ContactsSynchronization {
-  init(userPreferencesProvider: UserPreferencesProvider) {
-    self.userPreferencesProvider = userPreferencesProvider
+  private let userDefaults: UserDefaults
+  
+  init(userDefault: UserDefaults) {
+    self.userDefaults = userDefault
   }
```

Comment:
My issue with using UserDefaults directly is that this is not so nice to test, since it is directly manipulating the actual settings on the device (which is not what we want in tests). This is why I wrote the UserPreferenceProvider after the review of my changes, as you could essentially mock it away.

Maybe it's fine here, since this class anyway has to manipulate the device's contacts store directly as well which we surely would not want being done in a test, but I do think that this class has gotten sufficiently complex to warrant the underlying logic for updating contacts to be tested.

Alternatively, I would be satisfied with an issue being created to add the missing testing code if you do not want to hold back the release over this.

Reply:
Per @murilopereirame's request, I created #6479

Reply:
Since it's an objc class we can mock it anyway. I would be okay with wrapper if we couldn't mock it or if it was doing something useful